src/inference/src/os/win/win_system_conf.cpp

```diff
@@ -29,8 +29,6 @@ void CPU::init_cpu(CPU& cpu) {
         return;
     }
 
-    cpu._processors = GetMaximumProcessorCount(ALL_PROCESSOR_GROUPS);
-
     parse_processor_info_win(base_ptr,
```

Comment:
should we statically link hwloc in 2023.1 to avoid reinventing the wheel?

Reply:
I prefer the Windows API over adding additional libraries. And we always need to handle those corner cases when dependency library cannot report right information.

Let's validate the coverage of Windows API first. HWLOC is our backup plan if there are corner cases that the OS API cannot handle.